docs/resources/azure_locks.md

```diff
@@ -0,0 +1,114 @@
+| resource_group                 | Azure resource group that the targeted resource resides in. `MyResourceGroup`    |
+| resource_name                  | Name of the Azure resource on which the management locks are being tested. `MyVM` |
+| resource_type                  | Type of the Azure resource on which the management locks are being tested. `Microsoft.Compute/virtualMachines` |
+| resource_id                    | The unique resource ID of the Azure resource on which the management locks are being tested. `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Compute/virtualMachines/{vmName}` |
+
+Either one of the parameter sets can be provided for a valid query:
+- `resource_id`
+- `resource_group`, `resource_name` and `resource_type`
+- `resource_group`
+- None for a subscription level test.
+
+## Properties
+
+|Property       | Description                                                                          | Filter Criteria<superscript>*</superscript> |
+|---------------|--------------------------------------------------------------------------------------|-----------------|
+| ids           | A list of the unique resource ids of the management locks.                           | `id`            |
+| names         | A list of names of all the management locks being interrogated.                      | `name`          |
+| properties    | A list of properties for all the management locks being interrogated.                | `properties`    |
+
+<superscript>*</superscript> For information on how to use filter criteria on plural resources refer to [FilterTable usage](https://github.com/inspec/inspec/blob/master/dev-docs/filtertable-usage.md).
+
+## Examples
+
+### Check If a Specific Management Lock is Present for a Resource
+```ruby
+describe azure_locks(resource_group: 'rg-1', resource_name: 'my-VM', resource_type: 'Microsoft.Compute/virtualMachines') do
+  its('names')  { should include 'production_agents' }
+end
+```
+### Filters the Results to Include Only Those Management Locks which Include the Given Name
+```ruby
+describe azure_locks.where{ name.include?('production') } do
+  it { should exist }
+end
+```
+### Loop through All Virtual Machines to Test If They have Management Locks Defined on
+```ruby
+azure_virtual_machines.ids.each do |id|
+  describe azure_locks(resource_id: id) do
+    it { should exist }
+  end
+end
+``` 
+## Matchers
+
+This InSpec audit resource has the following special matchers. For a full list of available matchers, please visit our [Universal Matchers page](https://www.inspec.io/docs/reference/matchers/).
+
+### exists
+
+The control will pass if the filter returns at least one result. Use `should_not` if you expect zero matches.
+```ruby
+describe azure_locks(resource_group: 'rg-1', resource_name: 'my-VM', resource_type: 'Microsoft.Compute/virtualMachines') do
+  it { should exist }
+end
```

Comment:
This functionality would probably be more applicable on the singular resource. Usually, on the plural resource, we would provide a way to filter on the resources, rather than just pull back a specific resource. Perhaps there's a reason for this though?

Reply:
Management locks can be defined at the resource level. If this is the case, the resource identifier has to be provided. And, because there are multiple locks on a resource, the plural inspec resource is used. (+ backward compatibility)
https://docs.microsoft.com/en-us/rest/api/resources/managementlocks/listatresourcelevel

Reply:
Makes sense to include this for backward compatibility